src/index.ts

```diff
@@ -1,54 +1,17 @@
-function defaultGroqInferenceEndpoint(
-  groqApiKey?: string,
-  model: string = 'llama3-70b-8192'
-): (q: string) => Promise<string> {
-  return async (q: string) => {
-    const apiKey = groqApiKey;
-    const apiUrl = 'https://api.groq.com/openai/v1/chat/completions';
-    const requestBody = {
-      model,
-      messages: [
-        {
-          role: 'assistant',
-          content: 'Your answers must be consise',
-        },
-        {
-          role: 'user',
-          content: q,
-        },
-      ],
-      temperature: 0.1,
-    };
-    const response = await fetch(apiUrl, {
-      method: 'POST',
-      headers: {
-        'Content-Type': 'application/json',
-        Authorization: `Bearer ${apiKey}`,
-      },
-      body: JSON.stringify(requestBody),
-    });
-
-    if (!response.ok) {
-      throw new Error(`HTTP error! status: ${response.status}`);
-    }
-
-    const data = (await response.json()).choices[0].message.content;
-    return data;
-  };
-}
+type SystemPromptRole = 'assistant' | 'system';
 
-function defaultOpenAiChatGpt4Endpoint(
-  openAiApiKey?: string,
-  model: string = 'gpt-3.5-turbo'
+function defaultInferenceEndpoint(
+  apiKey?: string,
+  model: string = 'llama3-70b-8192',
+  apiUrl : string = 'https://api.groq.com/openai/v1/chat/completions',
+  systemPromptRole: SystemPromptRole = 'assistant'
 ): (q: string) => Promise<string> {
```

Comment:
I think 4 parameters might be just pushing the limit so it might be ok but otherwise I'm not super sure how I'd bring the number of parameters down without adversely affecting readability. Pulling some of those parameters into an object like your options type could reduce the parameters but I think that would be detrimental to readability.

However that does raise the issue of code style conventions. You might want to consider adding a linter of some kind to this project to ensure consistent code style.